src/containers/EmployeeList/Employees/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { Box, ChevronIcon, Pagination, Select, TextInput } from '@mantine/core';
+import { useEffect, useRef, useState } from 'react';
+import Employee from '../Employee';
+import styles from './styles.module.scss';
+import IEmployee from '@/types/employee';
+
+interface employeesProps {
+  listEmployees: Array<IEmployee>;
+  isValidating: boolean;
+  pageNumber: number;
+}
+export default function Employees({
+  listEmployees,
+  isValidating,
+  pageNumber,
+}: employeesProps) {
+  const [employeeList, setEmployeeList] = useState(listEmployees);
+  const [activePage, setPage] = useState(1);
+  const [search, setSearch] = useState('');
+  const [gender, setGender] = useState('all');
+  const [arrange, setArrange] = useState('all');
+  const typingTimeOut = useRef<any>(null);
+  const ascending = (a: String, b: String) => {
+    if (a > b) return 1;
+    if (a < b) return -1;
+    return 0;
+  };
+  const descending = (a: String, b: String) => {
+    if (a < b) return 1;
+    if (a > b) return -1;
+    return 0;
+  };
+  const handleChangeSearch = (event: any) => {
+    if (typingTimeOut.current) {
+      clearTimeout(typingTimeOut.current);
+    }
+    typingTimeOut.current = setTimeout(() => {
+      setSearch(event.target.value);
+    }, 600);
+  };
+  useEffect(() => {
```

Comment:
Inappropriate search

Reply:
you mean search without delay right?

Reply:
> you mean search without delay right?

Do you know why your search delay? I am not talking about debounce, it does not cause the delay

Reply:
Can you tell me which part is inappropriate in my search? i have tried some way but i didn't really realise what wrong with my search

Reply:
> Can you tell me which part is inappropriate in my search? i have tried some way but i didn't really realise what wrong with my search

You must answer this question, what does useEffect hook use for?

Reply:
now i know thank for your supports